packages-exp/auth-exp/src/core/user/user_impl.ts

```diff
@@ -317,3 +318,7 @@ export class UserImpl implements User {
     return user;
   }
 }
+
+export function userFromJSON(auth: externs.Auth, json: string): externs.User {
+  return UserImpl._fromJSON(_castAuth(auth), JSON.parse(json));
```

Comment:
Would be useful if this could take a plain object as well (some storage solutions parse the JSON automatically):

```javascript
const userObj = typeof json === "string" ? JSON.parse(json) : json;
return UserImpl._fromJSON(_castAuth(auth), userObj);
```